### PR TITLE
Update container-environment.md

### LINKED ISSUE #14127: Update container-environment.md to reflect reality
Forked from #12747. 

We need to clean up https://github.com/kubernetes/kubernetes/blob/master/docs/user-guide/container-environment.md. It's confusing to users.

* Ensure the description reflects current implementation
* Remove unimplemented features. This isn't a design doc.

cc @smarterclayton

### PR DESCRIPTION
**Pull request by bgrant0607**
Fixes #14127

### CODE REVIEW

File: docs/user-guide/container-environment.md [hunk 1]
```diff
@@ -89,51 +89,30 @@ Services have dedicated IP address, and are also surfaced to the container via D
...
-
-For hooks which have parameters, these parameters are passed to the event handler as a set of key/value pairs.  The details of this parameter passing is handler implementation dependent (see below).
+When a management hook occurs, the management system calls into any registered hook handlers in the container for that hook.  These hook handler calls are synchronous in the context of the pod containing the container. Typically we expect that users will make their hook handlers as lightweight as possible, but there are cases where long running commands make sense (e.g. saving state prior to container stop).
 
 ### Hook delivery guarantees
 
-Hook delivery is "at least one", which means that a hook may be called multiple times for any given event (e.g. "start" or "stop") and it is up to the hook implementer to be able to handle this
+Hook delivery is "at least once", which means that a hook may be called multiple times for any given event (e.g. "start" or "stop") and it is up to the hook implementer to be able to handle this
```
Comment (yujuhong) [inline docs/user-guide/container-environment.md]: I don't think this is entirely true. Consider a case where kubelet starts the container but dies before post-start hook is run, the restarted kubelet would not know that it needs to perform the post-start hook at all. This is what we *want* to do, so I guess it's fine.

Comment (bgrant0607) [inline docs/user-guide/container-environment.md]: I would like the doc to reflect reality. How about prestop hooks?

Comment (yujuhong) [inline docs/user-guide/container-environment.md]: Prestop hooks will be run at least once, as the document stated.

**Comment by bgrant0607** [comment]
Clarified lack of hook delivery guarantees. Please look at the second commit. I'll squash once it looks ok.

**Comment by yujuhong** [comment]
LGTM. Please squash, thanks.

**Comment by bgrant0607** [comment]
Squashed.

**Comment by bgrant0607** [comment]
@caesarxuchao I'd like to cherrypick this doc PR, but ran into a git problem and haven't had time to try again.

**Comment by caesarxuchao** [comment]
Ok. I'll do it.

**Comment by bgrant0607** [comment]
Thanks.

**Pull request merged**